qutebrowser/javascript/caret.js

```diff
@@ -1145,16 +1159,52 @@ window._qutebrowser.caret = (function() {
             }
         };
 
+    CaretBrowsing.reverseSelection = () => {
+        const sel = window.getSelection();
+        sel.setBaseAndExtent(
+            sel.extentNode, sel.extentOffset, sel.baseNode,
+            sel.baseOffset
+        );
+    };
+
+    CaretBrowsing.selectLine = function() {
+        const sel = window.getSelection();
+        sel.modify("extend", "right", "lineboundary");
+        CaretBrowsing.reverseSelection();
+        sel.modify("extend", "left", "lineboundary");
+        CaretBrowsing.reverseSelection();
+    };
+
+    CaretBrowsing.updateLineSelection = function(direction, granularity) {
+        if (!(granularity === "character") && !(granularity === "word")) {
+            window.
+                getSelection().
+                modify("extend", direction, granularity);
+            CaretBrowsing.selectLine();
+        }
+    };
+
+    CaretBrowsing.selectionEnabled = function() {
+        if (CaretBrowsing.selectionState === CaretBrowsing.SelectionState.NONE) {
+            return false;
+        }
+        return true;
```

Comment:
This function could probably just be `return CaretBrowsing.selectionState !== CaretBrowsing.SelectionState.NONE;

Reply:
Or alternatively, get rid of the function and use `selectionState !== CaretBrowsing.SelectionState.NONE` directly, like you've done in some (but not all) places below.

Reply:
Sorry I probably missed it somehow, I will fix it.

Reply:
No problem!